packages/server/lib/index.ts

```diff
@@ -41,16 +40,19 @@ export class Server {
     this.ENVIRONMENT = Container.get("ENVIRONMENT");
 
     this.bootKoa();
+    this.server = http.createServer(this.app.callback());
     bootLoader.addBootFunction(this.constructor.name, this.boot.bind(this));
+    gracefulShutdown.addServer(this.constructor.name, this.server);
   }
 
   private async boot(): Promise<void> {
     try {
       // start KOA on PORT
-      this.server = http.createServer(this.app.callback()).listen(this.serverConfig.port);
```

Comment:
Why have you moved this from the boot-function to constructor?

Is it necessary to run the terminus thing even before booting is finished?

Reply:
From my perspective, the constructor constructs everything that is needed and the boot function asynchronously boots and starts the things that have been constructed / created before. Thus, creating a server, configuring it and adding routes like `/healthcheck` is part of the constructing phase, whilst starting the server using `server.listen` belongs to the booting phase.

Do you disagree, @dustin-H ?

Reply:
I just wanted to understand.

I had a look at the Node.js specs. It is really just a constructor for that http.Server object, so its fine to do that inside the other constructor. :+1: